src/org/etools/j1939_84/controllers/part11/Part11Step08Controller.java

```diff
@@ -101,7 +99,7 @@ protected void run() throws Throwable {
         List<Integer> missingAddresses = determineNoResponseAddresses(dsResults, addresses);
         missingAddresses.stream()
                         .map(address -> getCommunicationsModule().requestDM20(getListener(), address))
-                        .forEach(dsResults::add);
+                        .peek(dsResults::add);
```

Comment:
You can’t change this to `.peek()`. The `.forEach()` is a terminal operation.  Without the terminal operation, the stream will never start emitting values.

Reply:
Already caught.